ingestion_server/test/unit_tests/test_distributed_reindex_scheduler.py

```diff
@@ -0,0 +1,87 @@
+import json
+from unittest import mock
+
+import pook
+import pytest
+
+from ingestion_server import distributed_reindex_scheduler
+
+
+@pytest.mark.parametrize(
+    "estimated_records, record_limit, workers, expected_ranges",
+    [
+        # One worker
+        (100, 1000, ["worker1"], [(0, 100)]),
+        # Multiple workers, even split
+        (100, 1000, ["worker1", "worker2"], [(0, 50), (50, 100)]),
+        # Multiple workers, uneven split
+        # NOTE: records get dropped in this case
+        (100, 1000, ["worker1", "worker2", "worker3"], [(0, 33), (33, 66), (66, 99)]),
+        # One worker, limited
+        (100, 55, ["worker1"], [(0, 55)]),
+        # Two workers, limited
+        (100, 50, ["worker1", "worker2"], [(0, 25), (25, 50)]),
+    ],
+)
+def test_assign_work(estimated_records, record_limit, workers, expected_ranges):
+    # Checks for the parameters
+    assert len(workers) == len(
+        expected_ranges
+    ), "Number of workers and expected ranges do not match, correct the test parameters"
+    # Set up database mock response
+    mock_db = mock.MagicMock()
+    mock_db.cursor.return_value.__enter__.return_value.fetchone.return_value = [
+        estimated_records
+    ]
+    # Enable pook & mock other internal functions
+    with pook.use(), mock.patch(
+        "ingestion_server.distributed_reindex_scheduler.get_record_limit"
+    ) as mock_get_record_limit, mock.patch(
+        "ingestion_server.distributed_reindex_scheduler._wait_for_healthcheck"
+    ) as mock_wait_for_healthcheck:
+        mock_wait_for_healthcheck.return_value = True
+        mock_get_record_limit.return_value = record_limit
+
+        post_mocks = [
+            pook.post(f"http://{worker}:8002/indexing_task") for worker in workers
+        ]
+
+        distributed_reindex_scheduler._assign_work(
+            mock_db,
+            workers,
+            "sample_model",
+            "sample_table",
+            "sample_index",
+        )
+
+        for worker, mock_post, (start_id, end_id) in zip(
+            workers, post_mocks, expected_ranges
+        ):
+            assert mock_post.calls > 0
+            data = json.loads(mock_post.matches[0].body)
+            print(f"{data=}")
+            assert data == {
+                "model_name": "sample_model",
+                "table_name": "sample_table",
+                "target_index": "sample_index",
+                "start_id": start_id,
+                "end_id": end_id,
+            }
```

Comment:
Rather than pulling the request data out of the pook mock, it's better to tell pook that's what to match in the first place:

```
pook.post(...).json({
  "model_name": "sample_model",
  "table_name": "sample_table",
  "target_index": "sample_index",
  "start_id": start_id,
  "end_id": end_id,
})
```

If pook sees a request that doesn't match the expected body, it will raise a no match exception, with a detailed diff showing why the actual request didn't match the expected request.

This also lets you avoid needing to think about pook's internals when pulling out the matched request data manually like this, and would remove the need for this complex secondary loop, which could be changed to just `[assert mock.done for mock in post_mocks]` or, better yet, `pook.done`, which checks that all mocks registered in context are exhausted.